src/interface/src/app/signup/signup.component.spec.ts

```diff
@@ -35,13 +35,52 @@ describe('SignupComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  describe('onSubmit', () => {
-    it('calls signup()', () => {
-      spyOn(component, 'signup').and.callThrough();
+  describe('signup form', () => {
+    it('disables continue button when first half of form is incomplete', () => {
+      expect(component.enableContinue()).toBeFalse();
+    });
+
+    it('enables continue button when first half of form is complete', () => {
+      component.form.get('firstName')?.setValue('Foo');
+      component.form.get('lastName')?.setValue('Bar');
+
+      expect(component.enableContinue()).toBeTrue();
+    });
+
+    it('disables submit when password fields are not equal', () => {
+      component.form.get('password1')?.setValue('foo');
+      component.form.get('password2')?.setValue('bar');
+
+      expect(component.form.valid).toBeFalse();
+    });
+
+    it('enables submit when all fields are valid', () => {
+      component.form.get('firstName')?.setValue('Foo');
+      component.form.get('lastName')?.setValue('Bar');
+      component.form.get('username')?.setValue('testuser');
```

Comment:
is the backend responsible for verifying that no two usernames are the same?

Reply:
That's right! The backend won't allow two users with the same username or email to be created and will return an error.